gitops/snapshot.go

```diff
@@ -268,6 +268,24 @@ func IsSnapshotValid(snapshot *applicationapiv1alpha1.Snapshot) bool {
 	return false
 }
 
+// IsSnapshotStatusConditionSet checks if the condition with the conditionType in the status of Snapshot has been marked as the conditionStatus and reason.
+func IsSnapshotStatusConditionSet(snapshot *applicationapiv1alpha1.Snapshot, conditionType string, conditionStatus metav1.ConditionStatus, reason string) bool {
+	condition := meta.FindStatusCondition(snapshot.Status.Conditions, conditionType)
+	if condition == nil && conditionType == AppStudioTestSuceededCondition {
+		condition = meta.FindStatusCondition(snapshot.Status.Conditions, LegacyTestSuceededCondition)
+	}
```

Comment:
Do we miss unittest coverage in lines below?

Reply:
It's added. Thanks

Reply:
But I still see codecov warnings about missing coverage

Reply:
https://app.codecov.io/gh/redhat-appstudio/integration-service/pull/263  patch coverage is just 30%

Reply:
minimum is 70% desired 80%, there is a huge code block uncovered

Reply:
The code in controllers/integrationpipeline/integrationpipeline_adapter.go#L129-L141 is in the end of the function  EnsureSnapshotPassedAllTests(), in the middle of the function, some mocked interactions are needed. Then it's hard to touch the end of the function. So, it's not covered.

Reply:
if the rest of the team waives coverage level, it's fine by me then

Reply:
It's 78.26% now

Reply:
Great work @jinqi7! I think we can conclude this discussion and move forward.